x-pack/plugins/security_solution/scripts/openapi/bundle.js

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+require('../../../../../src/setup_node_env');
+const { bundle } = require('@kbn/openapi-bundler');
+const { resolve } = require('path');
+
+const SECURITY_SOLUTION_ROOT = resolve(__dirname, '../..');
+
+bundle({
+  rootDir: SECURITY_SOLUTION_ROOT,
+  sourceGlob: './common/api/**/*.schema.yaml',
```

Comment:
> Not sure I understand why it ends up generating a single bundle for the whole solution.

I managed to produce one correct bundle eventually after renaming `ListRequestQuery`. Let me explain what conflicts we had

- `Page` defined in `x-pack/plugins/security_solution/common/api/detection_engine/model/pagination.schema.yaml` 
```
Page:
      type: integer
      minimum: 1
      description: Page number
```
and in `x-pack/plugins/security_solution/common/api/endpoint/model/schema/common.schema.yaml`
```
Page:
      type: integer
      default: 1
      minimum: 1
      description: Page number
```

In fact the first one is used in `PaginationResult` from `x-pack/plugins/security_solution/common/api/detection_engine/model/pagination.schema.yaml`  which in turn is used  in our internal API endpoint `/internal/detection_engine/rules/{ruleId}/execution/events` in `x-pack/plugins/security_solution/common/api/detection_engine/rule_monitoring/rule_execution_logs/get_rule_execution_events/get_rule_execution_events_route.schema.yaml`. Internal endpoints don't end up in the target spec.

- `ListRequestQuery` defined in `x-pack/plugins/security_solution/common/api/endpoint/metadata/list_metadata.schema.yaml` and in `x-pack/plugins/security_solution/common/api/endpoint/actions/list.schema.yaml` are two completely unrelated types. In fact `x-pack/plugins/security_solution/common/api/endpoint/actions/list.schema.yaml` should describe `EndpointActionListRequestSchema` in `x-pack/plugins/security_solution/common/api/endpoint/actions/list_route.ts` so renaming  `ListRequestQuery` to `EndpointActionListRequestQuery` looks logical.

General idea is that we must track conflicts and try to resolve them manually. It's really confusing to expose same name component schema in adjacent files while they represent two different entities and renaming will help to reduce ambiguity.


> For testing, I created two different schema files that each define two components with the same names.

Your test specs were merged as expected. Components aren't referenced and bundler don't do any checked on them. There was a problem with unused components ended up in the target spec but it has been fixed [here](https://github.com/elastic/kibana/pull/171526/commits/5caef2fc66ab45b317d1a1fb65245730a99a66b8). In case of referenced components conflicts an error will be shown.

There are still use cases when conflicts can be silently swollen. Bundler isn't able to catch a case If there is a non referenced component defined in a file with API endpoint and this API endpoint references another component with the same name from a different file.